proxy/src/control/mod.rs

```diff
@@ -32,22 +29,22 @@ pub struct Control {
     disco: Resolver,
 }
 
-pub struct Background {
-    disco: destination::background::Config,
-}
-
-pub fn new(dns_resolver: dns::Resolver, default_destination_namespace: String) -> (Control, Background)
+pub fn new(
+    dns_resolver: dns::Resolver,
+    default_destination_namespace: String,
+    host_and_port: HostAndPort,
+) -> (Control, impl Future<Item = (), Error = ()>)
 {
-    let (tx, rx) = self::destination::new(dns_resolver, default_destination_namespace);
+    let (tx, b) = self::destination::new(
```

Comment:
now that `b` isn't trivially constructed before our eyes, i'd use a more descriptive name here.  maybe `task`